tests/test_streaming.py

```diff
@@ -59,6 +59,203 @@ def test_dataloader_epoch_size_no_streams(local_remote_dir: Tuple[str,
+    convert_to_mds(out_root=remote1,
+                   dataset_name='sequencedataset',
+                   num_samples=200,
+                   size_limit=1 << 8)
+    # stream 2 has samples 600 and above. This lets us differentiate between the samples from each stream
+    convert_to_mds(out_root=remote2,
+                   dataset_name='sequencedataset',
+                   num_samples=300,
+                   offset=600,
+                   size_limit=1 << 8)
+
+    stream1 = Stream(local=local1, remote=remote1, proportion=stream_1_proportion)
+    stream2 = Stream(local=local2, remote=remote2, proportion=stream_2_proportion)
+
+    # Build StreamingDataset
+    dataset = StreamingDataset(streams=[stream1, stream2],
+                               shuffle=shuffle,
+                               batch_size=batch_size,
+                               shuffle_seed=seed,
+                               num_canonical_nodes=num_canonical_nodes,
+                               batching_method='stratified')
+
+    # Build DataLoader
+    dataloader = DataLoader(dataset=dataset,
+                            batch_size=batch_size,
+                            num_workers=num_workers,
+                            drop_last=drop_last)
+
+    # Ensure that the samples seen in each batch match what the user set.
+    total_proportions = stream_1_proportion + stream_2_proportion
+    stream_2_batch_part = round((stream_2_proportion / total_proportions) * batch_size)
+    stream_1_batch_part = batch_size - stream_2_batch_part
+    for batch in dataloader:
+        samples = batch['sample']
+        # Check if constructed batch is the correct size
+        assert len(samples) == batch_size
+        stream_1_samples = 0
+        stream_2_samples = 0
+        for sample in samples:
+            # stream 1 goes until 600, stream 2 is everything after
+            if sample < 600:
+                stream_1_samples += 1
+            else:
+                stream_2_samples += 1
+        # check that the batch is consistently composed of the correct number of samples from each stream
+        assert stream_1_samples == stream_1_batch_part
+        assert stream_2_samples == stream_2_batch_part
+
+
+@pytest.mark.parametrize('stream_2_size', list(range(1, 65, 10)))
+@pytest.mark.usefixtures('local_remote_dir')
+def test_stratified_batching_Exception(local_remote_dir: Tuple[str, str], stream_2_size: int):
+
+    local, remote = local_remote_dir
+    local1 = os.path.join(local, 'stream1')
+    local2 = os.path.join(local, 'stream2')
+    remote1 = os.path.join(remote, 'stream1')
+    remote2 = os.path.join(remote, 'stream2')
+
+    # With a batch size of 8, stream 1 of size 1000, and stream 2 anywhere between 1 and 65,
+    # we expect stream 2 to be too small to be included in each batch, which should raise ValueError.
+    stream_1_size = 1000
+    batch_size = 8
+
+    with pytest.raises(ValueError, match=f'Number of samples for stream*'):
```

Comment:
Minor suggestion: You should include the code inside`with` clause which generates that exception, not all the code. This way we ensure that the exception that we are testing are bounded by that method. For example, in this case, `convert_to_mds()`, `Stream` and `StreamingDataset` does not generate the above exception, hence, we can leave that out from `with pytest.raises()`

Reply:
done!